test/src/snapshot/mod.rs

```diff
@@ -60,220 +74,294 @@ pub(super) async fn run(filter_regex: Option<&regex::Regex>) -> Result<()> {
+                let _ = writeln!(snapshot, "{}", last_output.unwrap_or_default());
+            }
+            toml::Value::Table(map) => {
+                let ty = map["repeat"].as_str().unwrap();
+                let cmds = map["cmds"].as_array().unwrap();
+
+                match ty {
+                    "for-each-block" => {
+                        fn remove_block_from_file(contents: &str, block_name: &str) -> String {
+                            let block_regex = Regex::new(&format!("\\/\\* START {block_name} \\*\\/[.\\s\\S]+?END {block_name} \\*\\/")).unwrap();
+                            block_regex
+                                .replace_all(contents, |_: &Captures| -> String { String::new() })
+                                .to_string()
+                        }
+
+                        let path = PathBuf::from_str(root).unwrap().join("src/main.sw");
+                        let byte_contents = std::fs::read(&path).unwrap();
+                        let contents = String::from_utf8(byte_contents.clone()).unwrap();
+
+                        let mut blocks = BTreeSet::new();
+
+                        for capture in find_blocks_regex.captures_iter(&contents) {
+                            let name = capture.get(1).unwrap().as_str().to_string();
+                            blocks.insert(name);
+                        }
+
+                        for block in blocks.iter() {
+                            let _ = writeln!(snapshot, "> Block: {block}");
+
+                            let mut undo = UndoFiles::default();
+                            undo.contents.insert(path.clone(), byte_contents.clone());
+
+                            let mut new_contents = contents.clone();
+                            for remove_block in blocks.iter() {
+                                if remove_block != block {
+                                    new_contents =
+                                        remove_block_from_file(&new_contents, remove_block)
+                                            .to_string();
+                                }
+                            }
 
-                    let mut insta = insta::Settings::new();
-                    insta.set_snapshot_path(root);
-                    insta.set_prepend_module_to_snapshot(false);
-                    insta.set_omit_expression(true);
-                    let scope = insta.bind_to_scope();
-                    insta::assert_snapshot!("stdout", snapshot);
-                    drop(scope);
+                            let _ = std::fs::write(&path, new_contents);
+                            let _ = run_cmds(test_name, repo_root, root, cmds, snapshot);
+                        }
+                    }
+                    _ => todo!(),
                 }
-                stdout(&format!("{}/{root}", repo_root.display()), &snapshot);
-
-                Ok(())
-            })
-        })
-        .collect();
+            }
+            _ => todo!(),
```

Comment:
Instead of todos, can we have more explicit messages about what is or is not supported?

Reply:
Done.